src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -440,6 +442,11 @@ public boolean isAlive(String name) {
             }
         }
         procs.clear();
+        try {
+            FileUtils.deleteDirectory(Path.of(System.getProperty("java.io.tmpdir"), "agent-work-dirs").toFile());
```

Comment:
Extract to constant?

Reply:
Actually when I look at this, it could be deleting work dirs still in use by other rules or even other tests, which is wrong.